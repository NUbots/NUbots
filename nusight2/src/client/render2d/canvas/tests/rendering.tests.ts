import { describe, expect, it } from "vitest";

import { hexToRGB } from "../rendering";

describe("hexToRGB", () => {
  it("converts hex colours to rgb", () => {
    expect(hexToRGB("#AABBCC")).toEqual({
      r: 0xaa,
      g: 0xbb,
      b: 0xcc,
    });
  });

  it("throws if not a hex colour", () => {
    expect(() => hexToRGB("#FFF")).toThrowErrorMatchingSnapshot();
    expect(() => hexToRGB("red")).toThrowErrorMatchingSnapshot();
  });
});
