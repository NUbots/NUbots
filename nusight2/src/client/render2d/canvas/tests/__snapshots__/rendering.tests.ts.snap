// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`hexToRGB > throws if not a hex colour 1`] = `[Error: Color #FFF is not a hex color]`;

exports[`hexToRGB > throws if not a hex colour 2`] = `[Error: Color red is not a hex color]`;
